libs/autotagging-commons/src/main/java/org/opensearch/rule/storage/AttributeValueStoreFactory.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.rule.storage;
+
+import org.opensearch.autotagging.Attribute;
+import org.opensearch.autotagging.FeatureType;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+
+/**
+ * Factory class for AttributeValueStore
+ */
+public class AttributeValueStoreFactory {
+    private final static Map<String, AttributeValueStore<String, String>> attributeValueStores = new HashMap<>();
```

Comment:
If this map can be updated dynamically, safe to use a concurrentHashmap?

Reply:
Ideally the factory should be initialised at the node boot up time to load all attribute specific value stores. SInce changing the `AttributeValueStore` at runtime we are wiping the in-memory data in previous data structure.